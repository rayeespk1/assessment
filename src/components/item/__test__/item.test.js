import React from "react";
import '@testing-library/jest-dom'
import { render, cleanup } from "@testing-library/react";
import Item from "../item";

afterEach(cleanup);

test("renders", () => {
    const { asFragment } = render(<Item id="1" title="head title" status="pending" changeStatus={null} />);
    expect(asFragment()).toMatchSnapshot();
});

test("loading correct data", () => {
    const { getByTestId, getByText } = render(<Item id="1" title="head title" status="pending" changeStatus={null} />);

    expect(getByTestId("titleTag")).toHaveTextContent("head title");
    expect(getByTestId("statusTag")).toHaveTextContent("pending");
    expect(getByTestId("outerTag")).toHaveClass("item pending");
});