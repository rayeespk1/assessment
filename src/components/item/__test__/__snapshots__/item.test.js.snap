// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="item pending"
    data-testid="outerTag"
  >
    <p
      data-testid="titleTag"
    >
      head title
    </p>
    <p
      data-testid="statusTag"
    >
      pending
    </p>
  </div>
</DocumentFragment>
`;
